client_test.go

```diff
@@ -376,10 +378,44 @@ func Test_AccountResources(t *testing.T) {
 	assert.Greater(t, len(resourcesBcs), 0)
 }
 
+// A worker thread that reads from a chan of transactions that have been submitted and waits on their completion status
+func concurrentTxnWaiter(
+	results chan TransactionSubmissionResponse,
+	waitResults chan ConcResponse[*api.UserTransaction],
+	client *Client,
+	t *testing.T,
+	wg *sync.WaitGroup,
+) {
+	if wg != nil {
+		defer wg.Done()
+	}
+	responseCount := 0
+	for response := range results {
+		responseCount++
+		assert.NoError(t, response.Err)
+
+		waitResponse, err := client.WaitForTransaction(response.Response.Hash, PollTimeout(21*time.Second))
+		if err != nil {
+			t.Logf("%s err %s", response.Response.Hash, err)
+		} else if waitResponse == nil {
+			t.Logf("%s nil response", response.Response.Hash)
+		} else if !waitResponse.Success {
+			t.Logf("%s !Success", response.Response.Hash)
+		}
+		waitResults <- ConcResponse[*api.UserTransaction]{Result: waitResponse, Err: err}
+	}
+	t.Logf("concurrentTxnWaiter done, %d responses", responseCount)
+	// signal completion
+	// (do not close the output as there may be other workers writing to it)
+	waitResults <- ConcResponse[*api.UserTransaction]{Result: nil, Err: nil}
+}
+
 func Test_Concurrent_Submission(t *testing.T) {
-	const numTxns = uint64(10)
+	const numTxns = uint64(100)
+	const numWaiters = 4
+	netConfig := LocalnetConfig
 
-	client, err := NewClient(DevnetConfig)
+	client, err := NewClient(netConfig)
```

Comment:
You should be able to use NewTestClient, which handles which network in one place (for this it's all localnet)